server/plugin/hooks_test.go

```diff
@@ -0,0 +1,119 @@
+package plugin
+
+import (
+	"context"
+	"errors"
+	"reflect"
+	"testing"
+
+	"bou.ke/monkey"
+	"github.com/Brightscout/mattermost-plugin-servicenow-virtual-agent/server/serializer"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/plugin"
+	"github.com/mattermost/mattermost-server/v5/plugin/plugintest"
+	"github.com/stretchr/testify/mock"
+	"golang.org/x/oauth2"
+)
+
+func Test_MessageHasBeenPosted(t *testing.T) {
+	defer monkey.UnpatchAll()
+
+	for _, testCase := range []struct {
+		description                       string
+		Message                           string
+		getChannelError                   *model.AppError
+		getUserError                      error
+		parseAuthTokenError               error
+		sendMessageToVirtualAgentAPIError error
+	}{
+		{
+			description: "Message is posted and successfully sent to Virtual Agent",
+			Message:     "mockMessage",
+		},
+		{
+			description:     "Message is posted but failed to get current channel",
+			getChannelError: &model.AppError{},
+			Message:         "mockMessage",
+		},
+		{
+			description:  "Message is posted but failed to get user from KV store",
+			getUserError: errors.New("mockError"),
+			Message:      "mockMessage",
+		},
+		{
+			description:  "Message is posted but user is not connected to ServiceNow",
+			getUserError: ErrNotFound,
+			Message:      "mockMessage",
+		},
+		{
+			description: "Message is posted but user is not connected to ServiceNow",
+			Message:     "disconnect",
+		},
+		{
+			description:         "Message is posted but failed to parse auth token",
+			parseAuthTokenError: errors.New("mockError"),
+			Message:             "mockMessage",
+		},
+		{
+			description:                       "Message is posted but failed to parse auth token",
+			sendMessageToVirtualAgentAPIError: errors.New("mockError"),
+			Message:                           "mockMessage",
+		},
+	} {
+		t.Run(testCase.description, func(t *testing.T) {
+			p := Plugin{}
+
+			p.botUserID = "mock-botID"
+			mockAPI := &plugintest.API{}
+
+			mockAPI.On("LogError", mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.Anything, mock.AnythingOfType("string"), mock.AnythingOfType("string")).Return("LogError error")
```

Comment:
Use something like this - https://github.com/Brightscout/mattermost-plugin-moodle-sync/blob/940c0a0f02a016ce902f802d74872fce4c29f199/server/api_test.go#L34